bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/ShellyDevices.java

```diff
@@ -0,0 +1,612 @@
+    public static final String SHELLYDT_PLUS3EM63 = "S3EM-003CXCEU63";
+    public static final String SHELLYDT_PLUSBLUGW = "SNGW-BT01";
+    public static final String SHELLYDT_PLUSBLUGWG3 = "S3GW-1DBT001";
+
+    // Shelly Pro Series
+    public static final String SHELLYDT_PRO1 = "SPSW-001XE16EU";
+    public static final String SHELLYDT_PRO1_2 = "SPSW-101XE16EU";
+    public static final String SHELLYDT_PRO1_3 = "SPSW-201XE16EU";
+    public static final String SHELLYDT_PRO1PM = "SPSW-001PE16EU";
+    public static final String SHELLYDT_PRO1PM_2 = "SPSW-101PE16EU";
+    public static final String SHELLYDT_PRO1PM_3 = "SPSW-201PE16EU";
+    public static final String SHELLYDT_PRO1CB = "SPCB-01VENEU";
+    public static final String SHELLYDT_PRO2 = "SPSW-002XE16EU";
+    public static final String SHELLYDT_PRO2_2 = "SPSW-102XE16EU";
+    public static final String SHELLYDT_PRO2_3 = "SPSW-202XE16EU";
+    public static final String SHELLYDT_PRO2PM = "SPSW-002PE16EU";
+    public static final String SHELLYDT_PRO2PM_2 = "SPSW-102PE16EU";
+    public static final String SHELLYDT_PRO2PM_3 = "SPSW-202PE16EU";
+    public static final String SHELLYDT_PRO3 = "SPSW-003XE16EU";
+    public static final String SHELLYDT_PRO4PM = "SPSW-004PE16EU";
+    public static final String SHELLYDT_PRO4PM_2 = "SPSW-104PE16EU";
+    public static final String SHELLYDT_4PRO = "SHPSW04P";
+    public static final String SHELLYDT_PROEM50 = "SPEM-002CEBEU50";
+    public static final String SHELLYDT_PRO3EM = "SPEM-003CEBEU";
+    public static final String SHELLYDT_PRO3EM_2 = "SPEM-003CEBEU120";
+    public static final String SHELLYDT_PRO3EM3CT63 = "SPEM-003CEBEU63";
+    public static final String SHELLYDT_PRO3EM400 = "SPEM-003CEBEU400";
+
+    // Shelly Plus Mini Series
+    public static final String SHELLYDT_MINI_1 = "SNSW-001X8EU";
+    public static final String SHELLYDT_MINI_1G3 = "S3SW-001X8EU";
+    public static final String SHELLYDT_MINI_1G4 = "S4SW-001X8EU";
+    public static final String SHELLYDT_MINI_1PM = "SNSW-001P8EU";
+    public static final String SHELLYDT_MINI_1PMG3 = "S3SW-001P8EU";
+    public static final String SHELLYDT_MINI_1PMG4 = "S4SW-001P8EU";
+    public static final String SHELLYDT_MINI_PM = "SNPM-001PCEU16";
+    public static final String SHELLYDT_MINI_PMG3 = "S3PM-001PCEU16";
+    public static final String SHELLYDT_MINI_EM = "S4EM-001PXCEU16"; // PM was renamed to EM in Gen4
+
+    // Shelly BLU Series
+    public static final String SHELLYDT_BLUBUTTON = "SBBT";
+    public static final String SHELLYDT_BLUDW = "SBDW";
+    public static final String SHELLYDT_BLUMOTION = "SBMO";
+    public static final String SHELLYDT_BLUHT = "SBHT";
+
+    // Thing Type ID prefixes
+    public static final String THING_TYPE_SHELLYPLUS_PREFIX = "shellyplus";
+    public static final String THING_TYPE_SHELLYPRO_PREFIX = "shellypro";
+    public static final String THING_TYPE_SHELLYMINI_MIDDLE = "mini"; // format is shellyXXmini
```

Comment:
Nitpicking, "middle" seems a bit awkward. How about:
```suggestion
    public static final String THING_TYPE_SHELLYMINI_MATCH = "mini"; // format is shellyXXmini
```

or:
```suggestion
    public static final String THING_TYPE_SHELLYMINI_KEYWORD = "mini"; // format is shellyXXmini
```

focusing on what the string represents, not where it appears?

Reply:
yes, eliminated